### ISSUE TITLE
(lsp-plugins-fw) lsp-r3d-glx-lib.o not found when linking on Windows

### ISSUE THREAD

**Issue by Hanziness** [sadko4u/lsp-plugins#299]
Hello!

I believe I've got pretty far ahead with the compilation on Windows - thank you very much for your assistance with the previous compilation errors! I bumped into a linking issue that happens after compiling the plugins and returning to [lsp-plugin-fw](https://github.com/lsp-plugins/lsp-plugin-fw) to compile and link `respack`. On Windows, the linker wants to link `lsp-r3d-glx-lib` to it, but it was not built (yet?). 

These are the Linux build logs (Arch Linux build), where `lsp-r3d-glx-lib` is not included and it builds correctly:

```
2023-01-28T22:01:15.9831840Z echo "  g++  [lsp-plugin-fw] respack"
2023-01-28T22:01:15.9870157Z   g++  [lsp-plugin-fw] respack
2023-01-28T22:01:15.9871148Z mkdir -p /__w/lsp-plugins/lsp-plugins/.build/host/lsp-plugin-fw/utils-bin/
2023-01-28T22:01:15.9898869Z g++ -o /__w/lsp-plugins/lsp-plugins/.build/host/lsp-plugin-fw/utils-bin/respack /__w/lsp-plugins/lsp-plugins/.build/host/lsp-plugin-fw/lsp-plugin-fw-core.o /__w/lsp-plugins/lsp-plugins/.build/host/lsp-plugin-fw/util/common/checksum.o /__w/lsp-plugins/lsp-plugins/.build/host/lsp-plugin-fw/util/respack/respack.o /__w/lsp-plugins/lsp-plugins/.build/host/lsp-plugin-fw/util/respack.o -march=x86-64 -m64 -O2 -fvisibility=hidden -fno-exceptions -fno-rtti -fdata-sections -ffunction-sections -fno-asynchronous-unwind-tables -pipe -Wall -DLSP_INSTALL_PREFIX=\"/usr\"  /__w/lsp-plugins/lsp-plugins/.build/target/lsp-common-lib/lsp-common-lib.o  /__w/lsp-plugins/lsp-plugins/.build/target/lsp-dsp-lib/lsp-dsp-lib.o  /__w/lsp-plugins/lsp-plugins/.build/target/lsp-dsp-units/lsp-dsp-units.o  /__w/lsp-plugins/lsp-plugins/.build/target/lsp-lltl-lib/lsp-lltl-lib.o  /__w/lsp-plugins/lsp-plugins/.build/target/lsp-runtime-lib/lsp-runtime-lib.o            /__w/lsp-plugins/lsp-plugins/.build/target/lsp-r3d-base-lib/lsp-r3d-base-lib.o  /__w/lsp-plugins/lsp-plugins/.build/target/lsp-r3d-iface/lsp-r3d-iface.o  /__w/lsp-plugins/lsp-plugins/.build/target/lsp-tk-lib/lsp-tk-lib.o  /__w/lsp-plugins/lsp-plugins/.build/target/lsp-ws-lib/lsp-ws-lib.o              /__w/lsp-plugins/lsp-plugins/.build/target/lsp-plugins-shared/lsp-plugins-shared.o   -march=x86-64 -m64 -Wl,-z,relro,-z,now -Wl,--gc-sections              -lpthread  -ldl  -ljack   -lsndfile           -lX11   -lXrandr   -lcairo   -lfreetype         -lGL 
```

But in the Windows build it is included for linking and `ld` fails because `lsp-r3d-glx-lib.o` does not exist (indeed, it was not built previously) -- towards the end of the 6th line there is `/d/a/lsp-plugins/lsp-plugins/.build/target/lsp-r3d-glx-lib/lsp-r3d-glx-lib.o`. Also, if understand the library structure correctly, [`lsp-r3d-wgl-lib`](https://github.com/lsp-plugins/lsp-r3d-wgl-lib) should be used instead of `lsp-r3d-glx-lib` on Windows? It was not yet built, either. To me it seems the dependencies are correctly set in [dependencies.mk](https://github.com/lsp-plugins/lsp-plugin-fw/blob/master/dependencies.mk), so the issue could be elsewhere. This is the Windows build log (it compiled the plugins already -- *note* I specified only a subset of the plugins for now, which is why it finished with `lsp-plugins-para-equalizer`):

```
2023-01-29T20:24:31.1084196Z make[5]: Leaving directory '/d/a/lsp-plugins/lsp-plugins/modules/lsp-plugins-para-equalizer/src'
2023-01-29T20:24:31.1092217Z make[4]: Leaving directory '/d/a/lsp-plugins/lsp-plugins/modules/lsp-plugins-para-equalizer'
2023-01-29T20:24:31.1114265Z echo "  g++  [lsp-plugin-fw] respack.exe"
2023-01-29T20:24:31.1362203Z   g++  [lsp-plugin-fw] respack.exe
2023-01-29T20:24:31.1368550Z mkdir -p /d/a/lsp-plugins/lsp-plugins/.build/host/lsp-plugin-fw/utils-bin/
2023-01-29T20:24:31.1529142Z g++ -o /d/a/lsp-plugins/lsp-plugins/.build/host/lsp-plugin-fw/utils-bin/respack.exe /d/a/lsp-plugins/lsp-plugins/.build/host/lsp-plugin-fw/lsp-plugin-fw-core.o /d/a/lsp-plugins/lsp-plugins/.build/host/lsp-plugin-fw/util/common/checksum.o /d/a/lsp-plugins/lsp-plugins/.build/host/lsp-plugin-fw/util/respack/respack.o /d/a/lsp-plugins/lsp-plugins/.build/host/lsp-plugin-fw/util/respack.o -march=x86-64 -m64 -O2 -fvisibility=hidden -fno-exceptions -fno-rtti -fdata-sections -ffunction-sections -fno-asynchronous-unwind-tables -pipe -Wall -DLSP_INSTALL_PREFIX=\"/usr\"  /d/a/lsp-plugins/lsp-plugins/.build/target/lsp-common-lib/lsp-common-lib.o  /d/a/lsp-plugins/lsp-plugins/.build/target/lsp-dsp-lib/lsp-dsp-lib.o  /d/a/lsp-plugins/lsp-plugins/.build/target/lsp-dsp-units/lsp-dsp-units.o  /d/a/lsp-plugins/lsp-plugins/.build/target/lsp-lltl-lib/lsp-lltl-lib.o  /d/a/lsp-plugins/lsp-plugins/.build/target/lsp-runtime-lib/lsp-runtime-lib.o                        /d/a/lsp-plugins/lsp-plugins/.build/target/lsp-r3d-base-lib/lsp-r3d-base-lib.o  /d/a/lsp-plugins/lsp-plugins/.build/target/lsp-r3d-iface/lsp-r3d-iface.o  /d/a/lsp-plugins/lsp-plugins/.build/target/lsp-tk-lib/lsp-tk-lib.o  /d/a/lsp-plugins/lsp-plugins/.build/target/lsp-ws-lib/lsp-ws-lib.o      /d/a/lsp-plugins/lsp-plugins/.build/target/lsp-plugins-shared/lsp-plugins-shared.o  /d/a/lsp-plugins/lsp-plugins/.build/target/lsp-r3d-glx-lib/lsp-r3d-glx-lib.o -march=x86-64 -m64 -static-libgcc -static-libstdc++  -Wl,--gc-sections              -lshlwapi  -lwinmm  -lmsacm32  -lmpr  -lgdi32  -ld2d1  -lole32  -lwindowscodecs  -ldwrite  -luuid                
2023-01-29T20:24:31.4831456Z C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find D:/a/lsp-plugins/lsp-plugins/.build/target/lsp-r3d-glx-lib/lsp-r3d-glx-lib.o: No such file or directory
2023-01-29T20:24:31.5467916Z collect2.exe: error: ld returned 1 exit status
2023-01-29T20:24:31.5497220Z make[3]: *** [Makefile:867: /d/a/lsp-plugins/lsp-plugins/.build/host/lsp-plugin-fw/utils-bin/respack.exe] Error 1
2023-01-29T20:24:31.5498116Z make[3]: Leaving directory '/d/a/lsp-plugins/lsp-plugins/modules/lsp-plugin-fw/src'
```

Once again, the full build log is here: https://github.com/Hanziness/lsp-plugins/actions/runs/4038466593/jobs/6942482864#step:6:2011.

I'll keep looking in the meantime. I suspect the issue could be close to the Makefile. If I have some time, I'll try to see if adding `lsp-r3d-glx-lib` to the Windows dependencies, too, helps.

**Comment by sadko4u** [comment]
Hello!

`lsp-r3d-glx-lib` should not be used on Windows.

The `lsp-r3d-wgl-lib` should be used instead.

I didn't implement the complete build of LSP Plugins for windows yet.

**Comment by sadko4u** [comment]
The `dependencies.mk` were not adapted for Windows yet, so the build system is probably still trying to build the `lsp-r3d-glx-lib` instead of `lsp-r3d-wgl-lib`.

**Comment by sadko4u** [comment]
I think the reason is here:

https://github.com/lsp-plugins/lsp-plugin-fw/blob/987e5929a942e0976b01315675ada2700cd62e46/src/Makefile#L34-L46

In short: the build system merges the dependencies from the `lsp-plugin-fw` (since it requires it's own dependencies) and the dependencies of `lsp-plugin` package which explicitly require `lsp-r3d-glx-lib` at this moment.

Not `lsp-plugin-fw` nor `lsp-plugins` don't follow the rules of defining dependencies in the right manner at this moment (weren't reworked yet). The correct example of defining dependencies is, for example, the `lsp-tk-lib`:

https://github.com/lsp-plugins/lsp-tk-lib/blob/master/dependencies.mk

If you compare them with legacy style of LSP dependencies, you'll probably understand the idea why they were defined in such manner:

https://github.com/sadko4u/lsp-plugins/blob/master/dependencies.mk

**Comment by sadko4u** [comment]
@Hanziness FYI.
I've updated the dependencies. It should build better now.
But for VST2 there are still missing headers for VST2 plugin wrappers, so the build probably will fail on one of the latest stages.

**Comment by sadko4u** [comment]
The problem with library build has been solved now.

The rest job is to write the code that supports the core VST library loading for Windows.
But I believe this is the scope of #203 .

I'll mark this problem as 'Pending for release' and link to 1.2.6 release now.

All further work will be done within the issue #203.

**Comment by sadko4u** [comment]
@Hanziness I took your build configuration as a reference and made some improvements.
Now it is at least building:
https://github.com/sadko4u/lsp-plugins/actions/runs/4274380585/jobs/7440955984

**Comment by sadko4u** [comment]
Fixed in 1.2.6.
